source/ranecu.f90

```diff
@@ -73,6 +73,38 @@ subroutine ranecu(rvec,len,mcut)
 
 end subroutine ranecu
 
+! Init the random generator, that is, set seeds with proper value checks as described in [2]
+subroutine recuinit(is1,is2)
+
+  use crcoall
+
+  integer,           intent(in) :: is1
+  integer, optional, intent(in) :: is2
+
+  if(is1 < 1 .or. is1 > 2147483562) then
+    write(lout,"(a,i0)") "RANECU> ERROR Seed 1 must be an integer in the range 1 to 2147483562, got ", is1
+    call prror(-1)
+  else
+    iseed1 = is1
+  end if
+
+  if(present(is2)) then
+    if(is2 < 1 .or. is2 > 2147483398) then
+      write(lout,"(a,i0)") "RANECU> ERROR Seed 2 must be an integer in the range 1 to 2147483398, got ", is2
+      call prror(-1)
+    else
+      iseed2 = is2
+    end if
+  else
+    iseed2 = iseed1 + 19770404
+    if(iseed2 > 2147483398) then
+      ! Subtract max of seed 1 to ensure lower than max of seed 2
+      iseed2 = iseed2 - 2147483562
+    end if
+  end if
+
+end subroutine recuinit
+
 ! Set the seeds
```

Comment:
Add to the comment `(to be used for continuing a random number stream with seeds extracted using 'recuut()')`

Reply:
Yes, I'll add that. Good idea.